apigee/registry
# Implement API Linter Plugin in accordance to new Plugin Architecture

## Summary
In light of creating a method to specify style guides in the registry, it makes sense for us to have built-in support for API Linter. API-linter lints Protobuf specs according to the AIPs (https://aip.dev/), and was one of the main motivations for creating style guides in the registry in the first place.

This pull request implements the ability to specify style guides that contain rules from API linter, and have them be linted when computing conformance reports.

Furthermore, this follows the new plugin architecture that we've established for linters in the registry.

## Test
- Have registry running on `localhost:8080`.
- Ensure that https://github.com/googleapis/googleapis is extracted to your Desktop.
- Ensure that the following style guide is located under `~/styleguide.yaml`:

```
apg admin delete-project --name projects/protos

apg admin create-project --project_id protos --insecure --address localhost:8080

registry upload bulk protos --project-id protos ~/Desktop/googleapis --base-uri https://github.com/googleapis/googleapis/blob/$COMMIT

registry upload styleguide ~/styleguide.yaml --project-id=protos

registry compute conformance projects/protos/locations/global/apis/google-storage/versions/v1/specs/protos.zip --plugin=true

registry get projects/protos/locations/global/apis/google-storage/versions/v1/specs/protos.zip/artifacts/conformance-apilinterstyleguide --contents
```
The output on the command line should be the correct conformance report for the spec, which only reports the `core::0192::trademarked-names` which was specified in the style guide. For reference, ensure the output [matches the following gist.](https://gist.github.com/muhammadharis/70ea7523cc6770b27dec02bbe0d7f969)
